enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,333 @@
+Scorecard gathers any persisted volume data upon test completion and stores it in a local directory where scorecard is executed.  Users
+can specify a non-default local path for test output using a command flag as follows:
+```bash
+operator-sdk scorecard ./bundle --selector=suite=custom --test-output=/mytestoutput
+```
+
+#### Story 4
+A scorecard user can specify storage globally for any selected tests.
+The `storage` spec can be defined globally as in the following example:
+
+```yaml
+  storage: 
+    spec:
+      storageClassName: hostpath
+      capacity:
+        storage: 1Gi
+      accessModes:
+        - ReadWriteOnce
+      mountPath:
+        path: /test-output
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+```
+
+This would mean that any selected test would assume the global storage
+settings.  If a test has its own unique test storage defined, then those
+settings would override the global storage settings.
+
+### Implementation Details/Notes/Constraints
+
+This feature, if specified by a scorecard user, requires a storage class be defined and operational on the k8s cluster nodes that scorecard tests would be executed upon.  If no storage class is specified or a default storage class is not available an error is returned for the scorecard test.
+
+This feature follows a workflow as follows:
+ * scorecard allocates a unique PVC for a test if that test requests storage
+ * scorecard mounts that PVC into the test Pod so the test can write to
+a given mountPath any test output they want
+ * upon test completion and before cleaning up the tests, scorecard
+will create a container that mounts the PVC solely to harvest (exec/tar)
+the test output contents to the local host that scorecard is executing upon
+ * scorecard cleanup would include PVC cleanup as well as the current Pod cleanup
+ * the end user is left with a local directory of the harvested test output, with test output being divided into subdirectories based on suite and test names
+
+### Risks and Mitigations
+
+Scorecard users that don't require persistent storage are not impacted by this proposed feature.
+
+## Design Details
+
+### Unique Volume Claims
+Each scorecard test that specifies the storage feature will have a unique PersistentVolumeClaim created for that test.  This ensures that each
+test can be scheduled on any cluster node and that multiple tests will not co-mingle test output onto the same volume.  Scorecard deletes
+these volumes after test output has been gathered from the volume.  If the user specifies the command line flag of "--skip-cleanup", the volume will not
+be deleted.
+
+### Default Storage Values
+The following defaults will be supplied for the storage feature:
+ * 1Gi volume size
+ * RWO (ReadWriteOnce) access mode
+ * default storage class will be used
+ * /test-output mount path
```

Comment:
under what conditions are these defaults asserted? What is the minimum indication that a test or scorecard config wants storage?

Reply:
before when we were specifying labels, the presence of the storage label would cause storage to be provisioned, in this updated design, the presence of the storage spec in the config would cause storage to be provisioned.